### PR TITLE
Fixing Cross Runtime template

### LINKED ISSUE #553: [Bug] Cross-runtime library template does not compile properly
### Current behavior

See https://github.com/unoplatform/uno/discussions/15347#discussioncomment-8372607.

### Expected behavior

The library compiles properly

### Affected platforms

WebAssembly

### Template Host

dotnet new

### Host OS

Windows

### Template Options

`dotnet new unolib-crossruntime -o MyCrossRuntimeLibrary`

### Template Version

```shell
5.1.15
```

### Anything else we need to know?

_No response_

### LINKED ISSUE #559: Add Automated builds of all templates
## What would you like to be added:

The CI Build should include creating all of templates in the pack and ensuring that they build.

## Why is this needed:

We have no way of knowing currently if a breaking change is made to any templates outside of the unoapp template.

### PR DESCRIPTION
**Pull request by dansiegel**
GitHub Issue (If applicable): closes #

- closes #559
- closes #553

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Build or CI related changes

## What is the current behavior?

The Cross Runtime library for Wasm gets updated to use an Exe for the output type.

## What is the new behavior?

The Wasm head explicitly sets the Output Type as Library.

### CODE REVIEW

**Review by nickrandolph — APPROVED** [review]

**Comment by dansiegel** [comment]
@mergifyio backport release/stable/5.1

File: src/Uno.Templates/content/unolib-crossruntime/UnoCrossRuntimeLib/Directory.Build.props [hunk 1]
```diff
@@ -11,6 +11,19 @@
...
+    <RootNamespace>UnoCrossRuntimeLib</RootNamespace>
+    <AssemblyName>UnoCrossRuntimeLib</AssemblyName>
+    <OutputType>Library</OutputType>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <_DirectoryBuildPropsFile Condition="'$(_DirectoryBuildPropsFile)' == ''">Directory.Build.props</_DirectoryBuildPropsFile>
+    <_ParentDirectory>$([MSBuild]::NormalizePath($(MSBuildThisFileDirectory)\..))</_ParentDirectory>
+    <_ParentDirectoryBuildPropsBasePath Condition="'$(_ParentDirectoryBuildPropsBasePath)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(_ParentDirectory), '$(_DirectoryBuildPropsFile)'))</_ParentDirectoryBuildPropsBasePath>
+    <ParentDirectoryBuildPropsPath Condition="'$(_ParentDirectoryBuildPropsBasePath)' != '' and '$(_DirectoryBuildPropsFile)' != ''">$([System.IO.Path]::Combine('$(_ParentDirectoryBuildPropsBasePath)', '$(_DirectoryBuildPropsFile)'))</ParentDirectoryBuildPropsPath>
```
Comment (Youssef1313) [inline src/Uno.Templates/content/unolib-crossruntime/UnoCrossRuntimeLib/Directory.Build.props:24]: Looks like this is unnecessarily complex?

Shouldn't the following work?

```xml
<ParentDirectoryBuildPropsPath>$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))</ParentDirectoryBuildPropsPath>
```

File: build/templates/template-validation.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+- powershell: |
+      Set-PSDebug -Trace 1
+      $ErrorActionPreference = 'Stop'
+
+      cd UnoTemplateValidation1
+
+      if ('${{ parameters.templateName }}' -eq 'unoapp-uwp')
+      {
+          $msbuild = vswhere -latest -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe
+          $msbuild "/r /bl:$(build.artifactstagingdirectory)\$(Agent.JobName).binlog"
```
Comment (Youssef1313) [inline build/templates/template-validation.yml:61]:
```suggestion
          $msbuild /r /bl:$(build.artifactstagingdirectory)\$(Agent.JobName).binlog
```

**Pull request merged**